File: 45-jump-game-ii/jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int left=0;
        int right=0;
        int steps=0;
        int n= nums.length;

        while(right<n-1){
            int farthest=0;
            for(int i=left; i<=right; i++){
                farthest= Math.max(farthest, i+nums[i]);
            }
            left=right+1;
            right=farthest;
            steps+=1;
        }
        return steps;
    }
}

File: 45-jump-game-ii/jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:

        n= len(nums)
        left,right=0,0
        steps=0

        while(right<n-1):
            farthest=0
            for i in range(left,right+1):
                farthest=max(farthest,i+nums[i])
            left=right+1
            right=farthest
            steps+=1
        
        return steps
        
        